### PR TITLE
migrate from pkg_resources to importlib_resources

### PR DESCRIPTION
**Pull request by Tieqiong**

### CODE REVIEW

**Comment by Tieqiong** [comment]
I see the problem, will fix in a bit

File: src/diffpy/structure/tests/__init__.py [hunk 1]
```diff
@@ -38,10 +38,15 @@ def testsuite(pattern=""):
     from itertools import chain
     from os.path import dirname
 
-    from pkg_resources import resource_filename
+    # Use importlib.resources for Python 3.9 and newer or importlib_resources for older versions
+    try:
+        from importlib import resources as importlib_resources
```
Comment (sbillinge) [inline src/diffpy/structure/tests/__init__.py:43]: let's not import as `importlib_resources` which is deprecated.

let's do it the other way around, so use `resources` and `import importlib_resources as resources`

**Review by sbillinge — COMMENTED** [review]
looks good.  On quick comment.

File: src/diffpy/structure/tests/__init__.py (hunk 1 above)
Comment (Tieqiong) [reply src/diffpy/structure/tests/__init__.py]: @sbillinge fixed. Just import importlib_resources now

File: src/diffpy/structure/tests/__init__.py [hunk 2]
```diff
@@ -38,11 +38,7 @@ def testsuite(pattern=""):
     from itertools import chain
     from os.path import dirname
 
-    # Use importlib.resources for Python 3.9 and newer or importlib_resources for older versions
-    try:
-        from importlib import resources as importlib_resources
-    except ImportError:
-        import importlib_resources
+    import importlib_resources
```
Comment (sbillinge) [inline src/diffpy/structure/tests/__init__.py:41]: this doesn't seem to be the righ tbehavior for py > 3.9, no?   I am not sure if you understood my comment?

**Review by sbillinge — COMMENTED** [review]
please see comment.

I am not sure I understand what is happening here. Did you find a different fix?

File: src/diffpy/structure/tests/__init__.py (hunk 2 above)
Comment (Tieqiong) [reply src/diffpy/structure/tests/__init__.py:41]: I see the problem. I'm testing it in the wrong way. Let me change them

**Comment by Tieqiong** [comment]
@sbillinge I think it should be working now

File: src/diffpy/structure/tests/__init__.py (hunk 2 above)
Comment (sbillinge) [reply src/diffpy/structure/tests/__init__.py:41]: we don't want to use a backport.  We are trying to keep our code at the font edge, not the back edge.  That is my point....

**Review by sbillinge — COMMENTED** [review]
pls see my comment.

**Comment by Tieqiong** [comment]
Let me close this and open a clear PR

**Pull request closed without merging**